velox/exec/MergeJoin.h

```diff
@@ -20,6 +20,28 @@
 #include "velox/exec/Operator.h"
 
 namespace facebook::velox::exec {
+
+/// The merge join operator assumes both streams, left (from addInput()) and
+/// right (from rightSource), are sorted in the join key.
+///
+/// It works by identifying and maintaining a window of rows with key matches
+/// (leftMatch_ and rightMatch_), and outputting a cartesian product of each key
+/// match. Since keys can span multiple vectors, multiple batches from either
+/// side may need to be materialized and kept in memory. Therefore, the memory
+/// requirement is proportional to the size of the longest key match. Once all
+/// output for a particular key match is produced, the respective batches are
+/// discarded.
+///
+/// Output is produced outputBatchSize_ rows at a time.
+///
+/// The merge join operator generally returns dictionaries which are wrapped
+/// around input vectors. The output is aligned to left vectors, and since
+/// dictionaries cannot wrap around more than one vector, at times merge join
```

Comment:
> at times

Would that happen if join is selective, e.g. not every left row has a match?

Reply:
With the current code, it would return smaller batches in some cases, but they would all have dictionaries at least on the left. 

I suppose we could want to rebatch them in larger batches for efficiency in some cases - this is the same discussion as https://github.com/facebookincubator/velox/issues/7801